src/block_cache/block_chunkcache.c

```diff
@@ -261,10 +250,23 @@ __chunkcache_eviction_thread(void *arg)
  * __wt_chunkcache_get --
  *     If the cache has the data at the given size and offset, copy it into the supplied buffer.
  *     Otherwise, read and cache the chunks containing the requested data.
+ *
+ * During these operations we are holding one or more bucket locks. A bucket lock protects the
+ *     linked list (i.e., the chain) or chunks hashing into the same bucket. We hold the bucket lock
+ *     whenever we are looking for and are inserting a new chunk into that bucket. We must hold the
+ *     lock throughout the entire operation: realizing that the chunk is not present, deciding to
+ *     cache it, allocating the chunks metadata and inserting it into the chain. If we release the
+ *     lock during this process, another thread might cache the same chunk; we do not want that. We
+ *     insert the new chunk into the cache in the not valid state. Once we insert the chunk, we can
+ *     release the lock. As long as the chunk is marked as invalid, no other thread will try to
+ *     re-cache it or to read it. As a result, we can read data from the remote storage into this
+ *     chunk without holding the lock: this is what the current code does. We can even allocate the
+ *     space for that chunk outside the critical section: the current code does not do that. Once we
+ *     read the data into the chunk, we atomically set the valid flag, so other threads can use it.
```

Comment:
Nitpick: formatting for this paragraph is a little weird (All of the lines should have the same indent as the first).

Reply:
@fedorova to some more context to the indent guidelines.  The clang-format program with the options we use will not reformat lines that are indented.  Otherwise it will do line-filling and line-breaking.  The other thing is we usually try to have a one sentence summary, and we don't want the line broken for that.  Everything works out if we:
```
/*
 * __wt_my_function --
 *     This function does X and Y, and we indent this sentence four spaces.
 * More detail about how and why we do X and Y.   The formatter will fix up
 * these lines.  Rarely, I want to control formatting, so I can:
 *     1) Use an indent
 *     2) Stop indenting to return to auto-formatting.
 * More stuff that will be formatted.
 */
```

Reply:
Thanks for the tip, @ddanderson. I saved your note to my notebook and fixed all the comments to have the short first line.